### PR TITLE
[Property values] Add append_values method for modifying values

### PR DESCRIPTION
**Pull request by hkchekc**
Refer to Issue #382.

### CODE REVIEW

**Comment by jgrewe** [comment]
I guess we need a test case for that :)

File: nixio/property.py [hunk 1]
```diff
@@ -115,6 +115,15 @@ def _create_new(cls, nixparent, h5parent, name, dtype, oid=None):
 
         return newentity
 
+    def append_values(self, data):
+        """Method for appending values to existing values"""
+        arr = np.array(data).flatten('C')
+        ds = self._h5dataset.dataset
```
Comment (achilleas-k) [inline nixio/property.py]: H5DataSet exposes the `resize()` method through the `shape` property setter:
https://github.com/G-Node/nixpy/blob/ef2637cd198ef98b936e0c1211adbf76d8b57943/nixio/hdf5/h5dataset.py#L74

**Comment by achilleas-k** [comment]
I was wondering if there was a more pythonic way to do this and I realised that, since `values` is a property with a getter and a setter, anything we do on the property directly should just work, assuming the operation works on the type returned by the getter.

So this doesn't work:
```python
prop.values.append(10)
```
because `values` is a tuple and tuples don't have an append method, but this does
```python
prop.values += (10,)
```
because you can concatenate tuples with `+`.

It's essentially equivalent to
```python
v = prop.values
v += (10,)
prop.values = v
```
and our `values` setter already takes care of resizing when necessary.
https://github.com/G-Node/nixpy/blob/ef2637cd198ef98b936e0c1211adbf76d8b57943/nixio/property.py#L273

I suppose it might still be useful for users to have an `append_values()` method.  Though I think it could be simplified to just a `+` operation on the values property.  Perhaps we should also call it `extend()` if it's going to support multi-value append.  Could do both.

File: nixio/property.py [hunk 2]
```diff
@@ -115,6 +115,15 @@ def _create_new(cls, nixparent, h5parent, name, dtype, oid=None):
 
         return newentity
 
+    def append_values(self, data):
+        """Method for appending values to existing values"""
```
Comment (achilleas-k) [inline nixio/property.py]: The docstring shouldn't say what the method does, not describe what it is.

E.g., "Appends values to existing data."

File: nixio/property.py [hunk 3]
```diff
@@ -285,6 +276,38 @@ def values(self, vals):
...
+                raise TypeError("Array contains inconsistent values. "
+                                "Only values of type '{}' can be "
+                                "appended".format(vtype))
+
+        arr = np.array(data, dtype=vtype).flatten('C')
+        ds = self._h5dataset
+        src_len = len(self.values)
+        dlen = len(arr)
+        ds.shape = (src_len+dlen,)
+        ds.dataset.write_direct(arr, dest_sel=np.s_[src_len: src_len+dlen])
```
Comment (achilleas-k) [inline nixio/property.py]: We should probably expose the `write_direct()` method from the `h5py.Dataset` class through our `H5DataSet` wrapper.

Doing a `write_direct()` this way is good, more efficient than reading in the existing values, appending the new ones, and writing them back.  I don't like that we're duplicating the type checking code from the values setter property though.  We should probably pull those out and make a reusable private method.

**Pull request merged**